keymanagementscenarios.md

```diff
@@ -0,0 +1,40 @@
+Key management for container signing can be broadly categorized into three general use cases:
+- Signing (How are keys made available when Notary v2 signatures are generated?)
+- Trust store configuration (How do runtime environments determine which keys to trust?)
+- Key setup/revocation (How do key owners create/revoke keys? How do runtime environments get this information?)
+
+Signing Use Cases:
+- Developer runs docker build and notary v2 sign on local machine. Keys are stored in key store or plain text on local machine (Need to identify keystores we want to support. Should we preclude signing with plain text keys to improve security posture?).
+- Developer runs docker build and notary v2 sign on local machine. Keys are stored in a USB Token.
+- Developer runs docker build and notary v2 sign on local machine. Keys are stored in a network HSM.
+- Developer runs docker build and notary v2 sign on local machine. Keys are stored in a Cloud Key Management Service.
+- Automated build system runs docker build and notary v2 sign on build machine.
+- Automated build system runs docker build on a build machine. Signing is done on a second machine that verifies the container meets developer's release criteria.
+
+Trust Store Configuration:
+- Developer adds/removes their root public keys to the trust store used on a runtime host. The trust store will validate images pulled from any registry.
```

Comment:
What does “from any registry” mean here?
- Just that every image must be signed?
- Or that any “root” key can sign any image?

Neither is really attractive:
- It should be possible — it might even be the default! — for users to opt out and accept unsigned images from e.g. Docker Hub, while enforcing signatures from known vendors and their registries/repositories
- It should never be possible for one vendor to sign images that were supposed to come from some other vendor.

Or does it mean something else entirely?

Reply:
This applies to the runtime environment. It means that if the trusted root keys are defined, images from any registry can be verified as being signed with delegates belonging to the roots. I'll clarify the language in the next revision.